app/src/main/java/com/example/atheneum/viewmodels/TestAggregateViewModel.java

```diff
@@ -0,0 +1,72 @@
+package com.example.atheneum.viewmodels;
+
+import android.arch.core.util.Function;
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MediatorLiveData;
+import android.arch.lifecycle.Observer;
+import android.arch.lifecycle.Transformations;
+import android.arch.lifecycle.ViewModel;
+import android.support.annotation.Nullable;
+import android.support.v4.util.Pair;
+
+import com.example.atheneum.models.Book;
+import com.example.atheneum.models.User;
+import com.example.atheneum.utils.FirebaseQueryLiveData;
+import com.example.atheneum.viewmodels.FirebaseRefUtils.BooksRefUtils;
+import com.example.atheneum.viewmodels.FirebaseRefUtils.UsersRefUtils;
+import com.google.firebase.database.DataSnapshot;
+
+public class TestAggregateViewModel extends ViewModel {
+    private final FirebaseQueryLiveData borrowerQueryLiveData;
+    private final LiveData<User> borrowerLiveData;
+    private final FirebaseQueryLiveData bookQueryLiveData;
+    private final LiveData<Book> bookLiveData;
+
+    private final MediatorLiveData<Pair<User, Book>> borrowerBookPairLiveData;

```

Comment:
MediatorLiveData is a subclass of LiveData that listens for changes to other livedata streams and updates it's underlying data. It's basically what allows you to listen for changes to the User/Book and update the pair.